File: systems-for-design-and-implementation/labs/Java_ChatSockets/ChatServices/src/chat/services/IChatServer.java
package chat.services;

import chat.model.Message;
import chat.model.User;

/**
 * Created by IntelliJ IDEA.
 * User: grigo
 * Date: Mar 18, 2009
 * Time: 1:32:07 PM
 */
public interface IChatServer
{
    public void login(User user, IChatClient client) throws ChatException;

    public void sendMessage(Message message) throws ChatException;

    public void logout(User user, IChatClient client) throws ChatException;

    public User[] getLoggedFriends(User user) throws ChatException;
}

File: systems-for-design-and-implementation/labs/CSharp_ChatSockets/services/IChatServer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using chat.model;
namespace chat.services
{
    public interface IChatServer
    {
        void login(User user, IChatObserver client);
        void sendMessage(Message message);
        void logout(User user, IChatObserver client);
        User[] getLoggedFriends(User user);

    }
}
